natlas-server/app/api/routes.py

```diff
@@ -91,7 +91,7 @@ def submit():
 			response_body = json.dumps({"status": status_code, "message": "XML had too many hosts in it", "retry": False})
 
 		# If it's not an acceptable target, tell the agent it's out of scope
-		elif not current_app.ScopeManager.isAcceptableTarget(nmap.hosts[0].address):
+		elif len(nmap.hosts) == 1 and not current_app.ScopeManager.isAcceptableTarget(nmap.hosts[0].address):
```

Comment:
What happens if the agent provides > 1 hosts and the first one is out of scope? 

Unfortunately, Github mobile doesn't show what happens after this line

Reply:
Right now it will just get rejected as out of scope. But that should happen via the if statement before this change, where `if nmap.hosts_total != 1:`. Any properly running agents at this time should only be processing one host at a time and never sending xml files that contain multiple hosts. This is just a safeguard for that until we do support batch processing. The check for `len(nmap.hosts)` is also necessary instead of `nmap.hosts_total ==1` because `nmap.hosts_total` can be equal to 1 but if the host isn't up or has 0 ports open then `len(nmap.hosts)` is 0 and you can't access the first element.

It's hacky but this PR is just to fix a bug I inadvertently introduced.

Reply:
I appreciate the explanation. Def hacky, but its python, as long as you can read it, its good stuff.